src/activities/SelectGraphics.ts

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import Graphic from "@arcgis/core/Graphic";
+import GraphicsLayer from "@arcgis/core/layers/GraphicsLayer";
+import type {
+    IActivityHandler,
+    IActivityContext,
+} from "@geocortex/workflow/runtime/IActivityHandler";
+import { MapProvider } from "@geocortex/workflow/runtime/activities/arcgis/MapProvider";
+import { activate } from "@geocortex/workflow/runtime/Hooks";
+import MapView from "@arcgis/core/views/MapView";
+
+interface SelectGraphicsInputs {
+    /**
+     * @description A graphics layer that contains the graphics to be selected.
+     * @required
+     */
+    layer?: GraphicsLayer;
+
+}
+
+interface SelectGraphicsOutputs {
+    /**
+     * @description The selected graphics.
+     */
+    graphics: Graphic[] | undefined;
+}
+
+/**
+ * @category ArcGIS Maps SDK for JavaScript
+ * @description Allows the user to select graphics container in the provided Graphics Layer on the map.
+ * @clientOnly
+ * @supportedApps EXB, GWV
+ * */
+@activate(MapProvider)
+export default class SelectGraphics implements IActivityHandler {
+
+    async execute(inputs: SelectGraphicsInputs, context: IActivityContext,
+        type: typeof MapProvider): Promise<SelectGraphicsOutputs> {
+        const { layer } = inputs;
+        const mapProvider = type.create();
+        await mapProvider.load();
+        let pointerHandle: IHandle;
+        let clickHandle: IHandle;
+        let keyDown;
+        if (!mapProvider.view) {
+            throw new Error("view is required");
+        }
+        if (!layer) {
+            throw new Error("layer is required");
+        }
+        const mapView = mapProvider.view as MapView;
+
+        const graphics: Graphic[] | undefined = await new Promise((resolve) => {
+            pointerHandle = mapView.on("pointer-move", (event: __esri.ViewPointerMoveEvent) => {
+                void mapView.hitTest(event).then((hitResult: __esri.HitTestResult) => {
+                    const hit: boolean = hitResult.results.filter((result) => result.layer === layer).length > 0;
+                    if (hit) {
+                        mapView.container.style.cursor = "pointer";
```

Comment:
Explicitly assigning styles on DOM elements we don't control is not usually advised. Also our apps typically would use feature highlighting rather than the cursor style.

This isn't fully fleshed out, but what about something like this:
```
const layerView = await mapView.whenLayerView(layer);
let highlightHandle: __esri.Handle | undefined;
const graphics: Graphic[] | undefined = await new Promise((resolve) => {
    pointerHandle = mapView.on("pointer-move", (event: __esri.ViewPointerMoveEvent) => {
        void (async () => {
            const hitResult = await mapView.hitTest(event, {
                include: layer,
            });
            highlightHandle?.remove();
            const hit = hitResult.results.some((result) => result.layer === layer);
            if (hit) {
                const graphics = (hitResult.results as __esri.GraphicHit[]).map(x => x.graphic);
                highlightHandle = layerView.highlight(graphics)
            }
        })();
    });
```

I'm not positive on the proper clean up of the handles, but this is more of an Esri approved approach.

Reply:
Done